CommandLine/protos/resources/Sprite.proto

```diff
@@ -0,0 +1,41 @@
+syntax = "proto2";
+package buffers.resources;
+
+import "options.proto";
+
+message Sprite {
+  optional string name = 1;
+  optional int32 id = 2 [(gmx) = "GMX_DEPRECATED"];
+
+  optional bool preload = 4 [(gmx) = "GMX_DEPRECATED"];
+  optional bool transparent = 5 [(gmx) = "GMX_DEPRECATED"];
+  optional bool smooth_edges = 6 [(gmx) = "GMX_DEPRECATED"];
+
+  optional int32 alpha_tolerance = 7 [(gmx) = "coltolerance"];
+  optional bool separate_mask = 8 [(gmx) = "sepmasks"];
+  optional int32 origin_x = 9 [(gmx) = "xorig"];
+  optional int32 origin_y = 10 [(gmx) = "yorigin"];
+
+  enum Shape { PRECISE = 0; RECTANGLE = 1; DISK = 2; DIAMOND = 3; }
+
+  optional Shape shape = 11 [(gmx) = "colkind"];
+
+  enum BoundingBox { AUTOMATIC = 0; FULL_IMAGE = 1; MANUAL = 2; }
+
+  optional BoundingBox bbox_mode = 12 [(gmx) = "bboxmode"];
+
+  optional int32 bbox_left = 13;
+  optional int32 bbox_right = 14;
+  optional int32 bbox_top = 15;
+  optional int32 bbox_bottom = 16;
+
+  optional int32 type = 17;
+  optional int32 h_tile = 18 [(gmx) = "HTile"];
+  optional int32 v_tile = 19 [(gmx) = "VTile"];
+  optional int32 texture_group = 20 [(gmx) = "TextureGroups/TextureGroup0"];
+  optional bool for3D = 21 [(gmx) = "For3D"];

```

Comment:
What on earth does this mean? Generate scales and filters? Could you comment this, and possibly use a more descriptive name?

Reply:
"If your sprite is going to be used as a texture map for a 3D game, then you should check the Used for 3D box and the sprite will be given a texture page all of its own. Note: This will increase the texture memory needs of your game tremendously and so great care must be taken when using this option. Also note that all 3D textures must be a power of 2 (ie> 128x128, 256x256, 512x512 etc...)." <- according to gm manual

Reply:
It just means you are going to use it for a 3D model Josh, so it doesn't get added to a texture page so that you can use it with multitexturing. The reason you have to do this for 3D is because texture interpolation will cause other textures on the same texture page to bleed over on your model. However, if you are really good, Rusky says there's a way to do interpolation with paged 3D textures. I asked @rpjohnst about this for iPhone and Android games where computing resources are obviously limited for 3D graphics.